desmume/src/frontend/posix/gtk/main.cpp

```diff
@@ -1779,6 +1795,131 @@ static void PlayMovieDialog(GSimpleAction *action, GVariant *parameter, gpointer
     g_object_unref(pFileSelection);
 }
 
+static void ImportBackupMemoryDialog(GSimpleAction *action, GVariant *parameter, gpointer user_data)
+{
+    GtkFileFilter *pFilter_raw, *pFilter_ar, *pFilter_any;
+    GtkWidget *pFileSelection;
+    GtkWidget *pParent;
+    gchar *sPath;
+
+    if (desmume_running())
+        Pause(NULL, NULL, NULL);
+
+    pParent = GTK_WIDGET(pWindow);
+
+    pFilter_raw = gtk_file_filter_new();
+    gtk_file_filter_add_pattern(pFilter_raw, "*.sav");
+    gtk_file_filter_set_name(pFilter_raw, "Raw/No$GBA Save format (*.sav)");
+
+    pFilter_ar = gtk_file_filter_new();
+    gtk_file_filter_add_pattern(pFilter_ar, "*.duc");
+    gtk_file_filter_add_pattern(pFilter_ar, "*.dss");
+    gtk_file_filter_set_name(pFilter_ar, "Action Replay DS Save (*.duc,*.dss)");
+
+    pFilter_any = gtk_file_filter_new();
+    gtk_file_filter_add_pattern(pFilter_any, "*.sav");
+    gtk_file_filter_add_pattern(pFilter_any, "*.duc");
+    gtk_file_filter_add_pattern(pFilter_any, "*.dss");
+    gtk_file_filter_set_name(pFilter_any, "All supported types");
+
+    /* Creating the selection window */
+    pFileSelection = gtk_file_chooser_dialog_new("Import Backup Memory From ...",
```

Comment:
Please use [`GtkFileChooserNative`](https://developer.gnome.org/gtk3/unstable/gtk3-GtkFileChooserNative.html) instead, so that it can use the user’s preferred file picker instead of forcing the GTK one, plus it will work from inside sandboxes like Flatpak’s.

You can have a look around, for instance [`PlayMovieDialog()`](https://github.com/TASVideos/desmume/blob/master/desmume/src/frontend/posix/gtk/main.cpp#L1743) does pretty much the same thing.